linux/ibus-keyman/src/engine.c

```diff
@@ -745,25 +701,18 @@ process_end_action(IBusKeymanEngine *keyman) {
     // generated will be processed before the character we're adding. We need to send a
     // valid keyval/keycode combination so that it doesn't get swallowed by GTK but which
     // isn't very likely used in real keyboards. F24 seems to work for that.
-    ibus_engine_forward_key_event((IBusEngine*)keyman,
+    ibus_engine_forward_key_event(engine,
       KEYMAN_NOCHAR_KEYSYM,
       KEYMAN_F24_KEYCODE_OUTPUT_SENTINEL,
       (state & IBUS_RELEASE_MASK)
         ? IBUS_PREFILTER_MASK | IBUS_RELEASE_MASK
         : IBUS_PREFILTER_MASK);
   } else {
-    if (keyman->commit_item->char_buffer != NULL) {
-      commit_string(keyman, keyman->commit_item->char_buffer);
-      g_free(keyman->commit_item->char_buffer);
-      keyman->commit_item->char_buffer = NULL;
-    }
-    if (keyman->commit_item->emitting_keystroke) {
-      keyman->commit_item->emitting_keystroke = FALSE;
-      // We have an old ibus version without prefilter support, or a client that does support
-      // surrounding text. In either case we return FALSE because we emitted a keystroke
-      // so that the processing of the event will continue.
-      return FALSE;
-    }
+    // compliant app, or unpatched ibus
+    // If we have an old ibus version without prefilter support, or a client that does support
+    // surrounding text we already did everything that needs to be done including forwarding
+    // a keystroke. So we can return TRUE to stop processing of this event.
+      return TRUE;
```

Comment:
Does this mean finish_process_actions() never returns FALSE now?

Reply:
```suggestion
    return TRUE;
```

Reply:
> Does this mean finish_process_actions() never returns FALSE now?

yes. And this means the method doesn't have to return a bool. Changed. Thanks!